chaoslib/litmus/disk_fill_by_litmus.yml

```diff
@@ -0,0 +1,161 @@
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.spec.containers[?(@.name=="{{ app_container }}")].resources.requests.ephemeral-storage}' | grep -Eo '[[:alpha:]]+'
+  args:
+    executable: /bin/bash
+  register: req_value_storage
+
+- name:
+  include_vars: /experiments/generic/disk_fill/test_vars_diskfill.yml
+
+- name: Fetch the value of Request Storage Unit in KB
+  set_fact:
+    req_storage_KB: "{{ lookup('vars', req_value_storage.stdout) }}"
+
+- name: Fetch the value of Limit Storage Unit in KB
+  set_fact:
+    limit_storage_KB: "{{ lookup('vars', limit_value_storage.stdout)}}"
+
+- name: Multiply the request value with unit in KB
+  set_fact:
+    req_KB: "{{ req_value.stdout|int * req_storage_KB|int }}"
+
+- name: Multiply the limit value with unit in KB
+  set_fact:
+    limit_KB: "{{ limit_value.stdout|int * limit_storage_KB|int }}"
+
+- name: Convert the fill percent
+  set_fact: 
+    fill: "{{ fill_percent |float / 100 }}"
+
+- name: Fill percent into storage
+  set_fact:
+    to_fill: "{{ fill|float * limit_KB|float }}"
+
+- name: Get the Container ID
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} -o jsonpath='{.status.containerStatuses[?(@.name=="{{ app_container }}")].containerID}' | awk -F'//' '{print $2}'
+  args:
+    executable: /bin/bash
+  register: containerID
+
+- name: Identify the application node
+  shell: >
+    kubectl get pods {{ pod_name.stdout }} -n {{ namespace }} --no-headers -o custom-columns=:spec.nodeName
+  args:
+    executable: /bin/bash
+  register: app_node
+
+- name: Create the DaemonSet for Disk-Fill
+  shell: >
+    kubectl apply -f /experiments/generic/disk_fill/disk_fill_ds.yaml -n {{ namespace }}
+  args:
+    executable: /bin/bash
+  register: ds_status
+  
+- name: Confirm that the disk-fill daemonSet is running on all nodes
+  shell: >
+    kubectl get pods -l app=disk-fill --no-headers -o custom-columns=:status.phase -n {{ namespace }} | sort | uniq
+  args:
+    executable: /bin/bash
+  register: result
+  until: "result.stdout == 'Running'"
+  delay: 1
+  retries: 60
+
+- name: Record the disk-fill pod on app node
+  shell: >
+    kubectl get pods -l app=disk-fill -o wide -n {{ namespace }} | grep {{ app_node.stdout }} | awk '{print $1}'
+  args:
+    executable: /bin/bash
+  register: disk_fill_pod
+
+- name: Exec into the pod to get storage size
+  shell: >
+    kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ namespace }} -- bash -c "cd /diskfill && cd {{ containerID.stdout }} && du -h |tail -n1 | awk '{print $1}' | grep -Eo '+|[0-9]+'"
+  args:
+    executable: /bin/bash
+  register: pod_current_size
+
+- name: Exec into the pod to get storage unit
+  shell: >
+    kubectl exec -it {{ disk_fill_pod.stdout }} -n {{ namespace }} -- bash -c "cd /diskfill && cd {{ containerID.stdout }} && du -h |tail -n1 | awk '{print $1}' | grep -Eo '[[:alpha:]]+'"
+  args:
+    executable: /bin/bash
+  register: pod_current_storage_unit
```

Comment:
Similar to an earlier comment. Can we not process the output via ansible filters? However, it is your call on this.